### PR TITLE
Fix Unable to fetch logs from worker pod error in UI for k8s executor

### PR DESCRIPTION
**Pull request by snjypl**
_[ note: i have set `delete_worker_pods` to `False` so that the pod is not deleted after completion]_

for k8s executor while trying to view the task log from the UI.  we are getting the following error. 

```
airflow-webserver-5bf48475c-zdjxv
*** Trying to get logs (last 100 lines) from worker pod airflow-webserver-5bf48475c-zdjxv ***

*** Unable to fetch logs from worker pod airflow-webserver-5bf48475c-zdjxv ***
('Cannot find pod for ti %s', <TaskInstance: dataset_produces_2.producing_task_2 manual__2022-12-28T21:13:27.229615+00:00 [success]>)
```

i think, the issue is, while calling `PodGenerator.build_selector_for_k8s_executor_pod` we are passing `ti.try_number` instead of the `try_number` that was passed to the `_read` method. 

https://github.com/apache/airflow/blob/3ececb2c79307bd943aad116d7b0b5933af8185a/airflow/utils/log/file_task_handler.py#L167

https://github.com/apache/airflow/blob/3ececb2c79307bd943aad116d7b0b5933af8185a/airflow/utils/log/file_task_handler.py#L215-L222

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by ashb** [comment]
Paging @dstandish

**Comment by dstandish** [comment]
Looks reasonable to me, but needs a rebase

**Review by o-nikolas — APPROVED** [review]
lgtm, pending passing tests of course.

File: airflow/executors/base_executor.py [hunk 1]
```diff
@@ -356,13 +356,13 @@ def execute_async(
         """
         raise NotImplementedError()
 
-    def get_task_log(self, ti: TaskInstance) -> tuple[list[str], list[str]]:
+    def get_task_log(self, ti: TaskInstance, try_number: int) -> tuple[list[str], list[str]]:
         """
```
Comment (dstandish) [inline airflow/executors/base_executor.py:361]: wait this bit doesn't make sense to me.  if we already have the TI, why is it we also need to supply the try number?

**Review by dstandish — DISMISSED** [review]

File: airflow/executors/base_executor.py (hunk 1 above)
Comment (o-nikolas) [reply airflow/executors/base_executor.py:361]: There is some explanation in the code for `read`: https://github.com/apache/airflow/blob/c266aacf39a2f01cb3f91740f448934c7b3fd7be/airflow/utils/log/file_task_handler.py#L376-L410

`try_number` is plumbed in from `read` to `_read` and then should have been sent to the executors, but it was a miss:
https://github.com/apache/airflow/blob/c266aacf39a2f01cb3f91740f448934c7b3fd7be/airflow/utils/log/file_task_handler.py#L274-L315

**Comment by o-nikolas** [comment]
Anyone have more feedback for this one or shall I merge it?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
@dstandish ?

**Comment by o-nikolas** [comment]
Shall we just merge this fix then @potiuk @eladkal? Worst case we can always follow-up with more changes.

**Review by eladkal — APPROVED** [review]

**Comment by pierrejeambrun** [comment]
Conflicting, requires https://github.com/apache/airflow/pull/29482 and https://github.com/apache/airflow/pull/28161. Marking for 2.6.

**Pull request merged**